jaeger-core/src/main/java/io/jaegertracing/internal/JaegerTracer.java

```diff
@@ -590,8 +615,51 @@ public JaegerTracer build() {
             .withMetrics(metrics)
             .build();
       }
-      return new JaegerTracer(serviceName, reporter, sampler, registry, clock, metrics, tags,
-          zipkinSharedRpcSpan, scopeManager, baggageRestrictionManager, expandExceptionLogs);
+      return createTracer(
+          serviceName,
+          reporter,
+          sampler,
+          registry,
+          clock,
+          metrics,
+          tags,
+          zipkinSharedRpcSpan,
+          scopeManager,
+          baggageRestrictionManager,
+          expandExceptionLogs,
+          objectFactory);
```

Comment:
torn between this and just passing the Builder. It will need some accessors (maybe getters), but would reduce the toil of 10+ parameters.

Reply:
I 100% agree that this is beyond OK. In [Clean Code](http://a.co/bfhJe1V), Robert C. Martin suggests:

<img width="838" alt="clean-code-args" src="https://user-images.githubusercontent.com/2578297/43971409-eae51068-9c9e-11e8-80cd-982889dc3542.png">